matrix/lambdas/daemons/worker.py

```diff
@@ -41,15 +42,21 @@ def run(self, format: str, worker_chunk_spec: typing.List[dict]):
         self._parse_worker_chunk_spec(worker_chunk_spec)
         exp_dfs = []
         qc_dfs = []
-        for chunk_idx in range(len(self._bundle_uuids)):
-            dss_zarr_store = DSSZarrStore(bundle_uuid=self._bundle_uuids[chunk_idx],
-                                          bundle_version=self._bundle_versions[chunk_idx],
-                                          dss_instance=self._deployment_stage)
-            group = zarr.group(store=dss_zarr_store)
-            exp_df, qc_df = convert_dss_zarr_root_to_subset_pandas_dfs(
-                group, self._input_start_rows[chunk_idx], self._input_end_rows[chunk_idx])
-            exp_dfs.append(exp_df)
-            qc_dfs.append(qc_df)
+        with concurrent.futures.ThreadPoolExecutor(max_workers=8) as executor:
+            future_to_chunk_map = {executor.submit(self._read_chunk_from_DSS, chunk_idx): chunk_idx
+                                   for chunk_idx in range(len(self._bundle_uuids))}
+            for future in concurrent.futures.as_completed(future_to_chunk_map):
+                chunk_idx = future_to_chunk_map[future]
+                try:
+                    dss_zarr_store = future.result()
+                except Exception as e:
+                    print(f"Reading bundle uuid {self._bundle_uuids[chunk_idx]} from DSS caused exception {e}")
+                    continue
+                group = zarr.group(store=dss_zarr_store)
+                exp_df, qc_df = convert_dss_zarr_root_to_subset_pandas_dfs(
```

Comment:
I suspect it's going to be important to run `convert_dss_zarr_root_to_subset_pandas_dfs` in parallel. There are a lot of DSS requests that get made in there. Could you move that into the `_read` function?

Reply:
Oh totally missed this. Good catch